src/cli/cli_coap.cpp

```diff
@@ -214,8 +215,9 @@ otError Coap::ProcessSet(uint8_t aArgsLength, char *aArgs[])
 
     if (aArgsLength > 1)
     {
-        VerifyOrExit(strlen(aArgs[1]) < (kMaxBufferSize - 1), error = OT_ERROR_INVALID_ARGS);
-        strncpy(mResourceContent, aArgs[1], sizeof(mResourceContent) - 1);
+        VerifyOrExit(strlen(aArgs[1]) < sizeof(mResourceContent), error = OT_ERROR_INVALID_ARGS);
```

Comment:
```suggestion
        VerifyOrExit(strlen(aArgs[1]) < sizeof(mResourceContent) - 1, error = OT_ERROR_INVALID_ARGS);
```

Reply:
This isn't needed. The string length can be equal to `size - 1` without any issues.